db/commitlog/commitlog_entry.cc

```diff
@@ -36,6 +36,9 @@
 
 #include <seastar/core/simple-stream.hh>
 
+const detail::force_sync detail::force_sync::no;
+const detail::force_sync detail::force_sync::yes(db::timeout_clock::time_point{});
```

Comment:
time_point::min()? it's unlikely that 0 will be after the current time, but better to make it clear that we want yes to be the earliest time possible.

We might as well use the periodic timeout instead of no, and so fold periodic mode into this. We have too many paths.

Reply:
min is probably better, though I am pretty sure it is 0 in this (and most) case(s). Don't understand "use the periodic timeout" part? Fold what how?
Too many paths, yes. This particular part of the code is yet another bandaid in our somewhat disorganized attempts to decide what we actually need out of the module to make it work with all the changing requirements. Would be great to eventually simplify. But the "no" case is literally equivalent to the old "false" case, so I don't really see that it adds any "new" complexity. The non yes/no does somewhat, but fairly benignly imho.

Reply:
Let's see what sync modes we have now:
1. sync now no matter what
2. sync when we have 128k worth of buffer, or 10s elapsed, whichever came sooner
3. sync when we have min_alignment, or writer->sync elapsed

I was thinking of folding 2 and 3, and specify 1 in terms of 2/3 (timeout == time_point::min(), so always elapsed).

Reply:
Problem perhaps is that (1) is the batch mode, which has its own weird rules, and does not use/force the timer task. D you mean to move this into it simply forcing the timer to run? Does that not potentially add latency for the batch case?
Moreover, batch has its own rules, and when in place, does never even use the timer path. So should it remain such?

I am somewhat unsure on the term "fold" here. (2) and (3) are just two ways the timer/perioding sync is issued, except in (2) it is also done by the allocating code path directly. Is your suggestion to change _all_ paths that need to sync things to make the timer run? 
Since we only use one segment at a time, this is should be ok:ish, we won't really force any premature sync/flush-to-disk on segments (if we do, they are already non-allocating, and won't add any wasted space because of it - just ensure disk flushing).